pkg/resources/stream_on_directory_table_acceptance_test.go

```diff
@@ -0,0 +1,342 @@
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.name", id.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.database_name", id.DatabaseName())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.schema_name", id.SchemaName())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner", snowflakeroles.Accountadmin.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.comment", "bar")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.table_name", stage.ID().Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.source_type", string(sdk.StreamSourceTypeStage))),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.base_tables.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.base_tables.0", stage.ID().Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.type", "DELTA")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.stale", "false")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.mode", string(sdk.StreamModeDefault))),
+					assert.Check(resource.TestCheckResourceAttrSet(resourceName, "describe_output.0.stale_after")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner_role_type", "ROLE")),
+				),
+			},
+			// import
+			{
+				Config:       config.FromModel(t, modelWithExtraFieldsModified),
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedStreamOnDirectoryTableResource(t, resourceId).
+						HasNameString(id.Name()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasStageString(stage.ID().Name()).
+						HasCommentString("bar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnDirectoryTable_CopyGrants(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_stream_on_directory_table.test"
+
+	var createdOn string
+
+	stage, cleanupStage := acc.TestClient().Stage.CreateStageWithDirectory(t)
+	t.Cleanup(cleanupStage)
+
+	model := model.StreamOnDirectoryTable("test", id.DatabaseName(), id.Name(), id.SchemaName(), stage.ID().FullyQualifiedName())
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnTable),
```

Comment:
Hmm the test only checks if the resource was recreated and with the AlwaysIgnore suppress diff I think `copy_grants` doesn't work properly (because it's not changed). You can just assert the state and see the value is the same because of the diff suppress (I think).

Reply:
The point of this test is to make sure that changing copy_grants doesn't recreate the whole resource.

Reply:
Got it, I was wondering if the state value of copy_grants is even assertable since there's this IgnoreAfterCreate diff suppress that is keeping it always the same in state.

Reply:
I changed this a bit and added more tests.